jax/_src/pallas/core.py

```diff
@@ -310,9 +313,11 @@ def _convert_block_spec_to_block_mapping(
     return None
   if block_spec.index_map is None:
     compute_index = lambda *args, **kwargs: (0,) * len(aval.shape)
-    block_shape = aval.shape
   else:
     compute_index = block_spec.compute_index
+  if block_spec.block_shape is None:
+    block_shape = aval.shape
```

Comment:
That doesn't make a lot of sense either. A block spec should always contain both fields